lib/routes/account.js

```diff
@@ -1043,16 +1128,18 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
+
+        if (!sessionToken.emailVerified) {
+          // Send email verification
+          if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
+            return reply({})
+          }
+
+          return customs.check(
+            request.app.clientAddress,
+            sessionToken.email,
+            'recoveryEmailResendCode')
+            .then(mailer.sendVerifyCode.bind(
               mailer,
               sessionToken,
               sessionToken.emailCode,
```

Comment:
> 3. See the "you must verify your account" error message, hit "reset email".
> 4. Receive the verification email, click through it

I'm confused, where is "you must verify your account" and where is "reset email"? In the content server, or the browser? Is "reset" supposed to be "resend"? 

If so, at step 2, a verification email will be sent, a new sessionToken will be created, and that sessionToken will be sent in a login message to the browser. If the browser calls /recovery_email/resend_code with the sessionToken from step 2, I imagine the auth server sends an identical email with the same code?